src/main/java/com/direwolf20/buildinggadgets/common/integration/IBlockProviderItemHandler.java

```diff
@@ -0,0 +1,26 @@
+package com.direwolf20.buildinggadgets.common.integration;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+public interface IBlockProviderItemHandler extends IItemHandler {
+
+	/**
+	 * Returns the amount of a certain ItemStack that this IItemHandler currently has
+	 * @param toCount The ItemStack to count
+	 * @param player The player
+	 * @return The count of this item in the IItemHandler
+	 */
+	public int getItemCount(ItemStack toCount, EntityPlayer player);
+
+	/**
+	 * Extracts the requested amount (or less) of the specified item from the IItemHandler.
+	 * @param toExtract The ItemStack to extract
+	 * @param maxCount The maximum count that should be extracted
+	 * @param player The player
+	 * @return The amount that was actually extracted
+	 */
+	public int extractItems(ItemStack toExtract, int maxCount, EntityPlayer player);
```

Comment:
```suggestion
	public default int extractItems(EntityPlayer player, ItemStack toExtract, int maxCount, boolean simulate) {
	    int amountSaturated = 0;
	    for (int i = 0; i < getSlots(); i++) {

            ItemStack containerItem = inventory.getStackInSlot(i);

            if (containerItem.getItem() == target.getItem() && containerItem.getMetadata() == target.getMetadata()) {
                ItemStack stack = inventory.extractItem(i, amountRequired, simulate);
                amountSaturated += stack.getCount();

            }


            // Don't continue to check if we've saturated the amount.
            if( amountSaturated >= amountRequired )
                break;

        }
        
        return amountSaturated;
	}
```

Reply:
Furthermore I'd kind of prefer it to have extractItems with a simulate boolean instead of the getItemCount Method. Especially as there might be inventory which have a different amount of Items, then can be extracted (locked slots for example).

Reply:
I'm not sure that getItemCount could be replaced by having an extra simulate boolean. I reckon that if simulate is true, we would still expect to get an exact count of how many items it should *simulate* to extract. However, in the place where getItemCount is called currently, we don't know about how many items we want to extract.
Unless of course, we augment the javadoc to clearly state that if simulate is true, no exact value must be provided which should be simulated. But I think that isn't really conventional and would cause unneccessary confusion.

Reply:
getItemCount is currently used for the build preview so I'd say it should be as exact, as it can get.
Also weakening the simulate contract like this sounds reasonable to me, because you'd get the exact value anyway, once you extracted...

Reply:
What about instead renaming the `getItemCount` Method to something like `getItemsForExtraction` to signify what it shall report? That would be nicer then having a boolean argument telling the Method what it should return...